### PR TITLE
Feature/casava 1.8

### PR DESCRIPTION
**Pull request by byb121**
supports casava format 1.8

### CODE REVIEW

File: cgp_seq_input_val/fastq_read.py [hunk 1]
```diff
@@ -38,25 +46,41 @@ def __init__(self, fq_fh, line_no_in, curr_line):
...
         self.qual = qual
         self.file_pos = (line_no_in, line_no)
         self.last_line = curr_line  # as we need to pass this back
         self.name = None
-        self.end = None
+        self.pair_member = None
 
     def __str__(self):
-        return "%s\n%s\n+\n%s" % (self.header, self.seq, self.qual)
+        # TODO needs to output casava 1.8 format headers
```
Comment (keiranmraine) [inline cgp_seq_input_val/fastq_read.py]: Why is this TODO here?

File: cgp_seq_input_val/fastq_read.py [hunk 2]
```diff
@@ -68,16 +92,39 @@ def validate(self, filename):
         Raises:
             SeqValidationError - Generic errror with validation
         """
-        match = re.match(r'@(\S+)/([12])', self.header)
+        match = CASAVA_FASTQ_HEADER_PATTERN.match(self.seq_header)
         if match is None:
-            raise SeqValidationError("Sequence record header must begin with \
-                                     '@' one non-whitespace character and \
-                                     '/[12]', line %d of %s"
+            raise SeqValidationError("Sequence record header must match patten: '%s', line %d of %s"
```
Comment (keiranmraine) [inline cgp_seq_input_val/fastq_read.py]: `patten` spelling error

File: cgp_seq_input_val/seq_validator.py [hunk 3]
```diff
@@ -311,3 +329,9 @@ def setup_progress(self):
         print("Progress is %d's of record pairs" % (self.progress_pairs), file=sys.stderr)
         bar.update(0)
         return bar
+
+
+def get_fq_format(file_h):
+    read = FastqRead(file_h, 0, None)
+    file_h.seek(0)  # reset file pointer to the begaining
+    return read.get_fq_format()
```
Comment (keiranmraine) [inline cgp_seq_input_val/seq_validator.py]: Can you configure your editing tools to ensure you have a new line at the end of text files please?

**Review by keiranmraine — CHANGES_REQUESTED** [review]
Some minor corrections required.

File: cgp_seq_input_val/seq_validator.py (hunk 3 above)
Comment (keiranmraine) [reply cgp_seq_input_val/seq_validator.py]: Can see the issue if you view the main diff.

Comment (byb121) [reply cgp_seq_input_val/seq_validator.py]: Set in my vscode now.

**Comment by keiranmraine** [comment]
Requires manual merge, will complete on command line.  Otherwise all good.

**Pull request closed without merging**